util/microservice-sdk/js/test/unit/kafka/Consumer.js

```diff
@@ -396,58 +406,90 @@ describe('Refresh subscription', () => {
     expect(consumer.consumer.subscribe).toHaveBeenCalledWith([topic, regExpTopic]);
   });
 
-  test('Refresh subscription with retry', async (done) => {
+  test('Refresh subscription with retry', (done) => {
     // sets the fake timer to analyze the refreshSubscriptions
     jest.useFakeTimers();
 
-    // test subject
-    const consumer = new ConsumerBackPressure({});
-
-    consumer.holdOffTimeSubscription = 1000;
-    consumer.holdOffFactorTimeSubscription = 2.0;
-    consumer.maxHoldOffTimeSubscription = 3000;
+    const consumer = new Consumer(
+      {
+        'subscription.backoff.min.ms': 1000,
+        'subscription.backoff.max.ms': 10000,
+        'subscription.backoff.delta.ms': 1000,
+      },
+    );
 
     consumer.consumer = new KafkaMock.KafkaConsumer();
     consumer.isReady = true;
     const topic = 'tenant/dojot';
     consumer.topicMap[topic] = [
       { id: 'entry', callback: jest.fn() },
     ];
-    consumer.refreshSubscriptions(0);
+
+    consumer.consumer.subscribe = jest.fn();
+
+    // subscriptionProcedure(/*retries*/ 0) -> fails
```

Comment:
Is this commented code necessary?

Reply:
The idea of this sequence of comments is to explain what the expected behavior is. I'll try to make it clearer.